tiledb/sm/cpp_api/ndrectangle.h

```diff
@@ -246,26 +255,6 @@ class NDRectangle {
     return {std::move(start_str), std::move(end_str)};
   }
 
-  /**
-   * Retrieves a range for a given variable length string dimension index and
-   * range id.
-   *
-   * @param dim_name The dimension name.
-   * @param range_idx The range index.
-   * @return A pair of the form (start, end).
-   */
-  std::array<std::string, 2> range(const std::string& dim_name) {
```

Comment:
Using ordinary specialization syntax for this function based on old ll. 277-278 would work just fine. There's no need for the `tuple_ret` structure.
```
template<>
std::array<std::string, 2> range<std::string>(const std::string& dim_name)
```

Reply:
I pushed a commit with only a classic specialization, no struct. I remember either compiler wouldn't let me have that specialization on the return type, or it would select the incorrect implementation, that's why I added the struct, unfortunately I don't remember which one it was. At this latest version my local compiler both compiles fine and selects the correct implementation with your suggestion ^, let's see if build is green in the CI.